test/management_tests/procedures_test.py

```diff
@@ -594,3 +594,152 @@ async def test_nm_individual_address_write_address_found_other_in_programming_mo
+    )
+
+    await time_travel(management.management.MANAGAMENT_CONNECTION_TIMEOUT)
+
+    assert xknx.cemi_handler.send_telegram.call_args_list == [
+        call(read_address),
+    ]
+
+    assert not await task
+
+
+async def test_nm_individual_address_serial_number_write(time_travel):
+    """Test nm_individual_address_serial_number_write."""
+
+    xknx = XKNX()
+    xknx.cemi_handler = AsyncMock()
+    serial_number = b"aabbccddeeff"
+    individual_address = IndividualAddress("1.1.5")
+
+    task = asyncio.create_task(
+        procedures.nm_individual_address_serial_number_write(
+            xknx=xknx, serial=serial_number, individual_address=individual_address
+        )
+    )
+
+    write_address = Telegram(
+        destination_address=GroupAddress("0/0/0"),
+        payload=apci.IndividualAddressSerialWrite(
+            serial=serial_number, address=individual_address
+        ),
+    )
+    read_address = Telegram(
+        destination_address=GroupAddress("0/0/0"),
+        payload=apci.IndividualAddressSerialRead(serial=serial_number),
+    )
+    address_reply = Telegram(
+        source_address=individual_address,
+        destination_address=GroupAddress("0/0/0"),
+        direction=TelegramDirection.INCOMING,
+        payload=apci.IndividualAddressSerialResponse(
+            address=individual_address, serial=serial_number
+        ),
+    )
+
+    await time_travel(0)
+
+    await time_travel(0.5 * management.management.MANAGAMENT_ACK_TIMEOUT)
+    xknx.management.process(address_reply)
+
+    assert xknx.cemi_handler.send_telegram.call_args_list == [
+        call(write_address),
+        call(read_address),
+    ]
+
+    await time_travel(management.management.MANAGAMENT_CONNECTION_TIMEOUT)
+
+    await task
+
+
+async def test_nm_individual_address_serial_number_write_fail(time_travel):
+    """Test nm_individual_address_serial_number_write."""
+
+    xknx = XKNX()
+    xknx.cemi_handler = AsyncMock()
+    serial_number = b"aabbccddeeff"
+    individual_address = IndividualAddress("1.1.5")
+
+    task = asyncio.create_task(
+        procedures.nm_individual_address_serial_number_write(
+            xknx=xknx, serial=serial_number, individual_address=individual_address
+        )
+    )
+
+    write_address = Telegram(
+        destination_address=GroupAddress("0/0/0"),
+        payload=apci.IndividualAddressSerialWrite(
+            serial=serial_number, address=individual_address
+        ),
+    )
+    read_address = Telegram(
+        destination_address=GroupAddress("0/0/0"),
+        payload=apci.IndividualAddressSerialRead(serial=serial_number),
+    )
+
+    await time_travel(0)
+    await time_travel(management.management.MANAGAMENT_CONNECTION_TIMEOUT)
+
+    assert xknx.cemi_handler.send_telegram.call_args_list == [
+        call(write_address),
+        call(read_address),
+    ]
+
+    with pytest.raises(ManagementConnectionError):
+        await task
```

Comment:
```suggestion
    await time_travel(0)
    assert xknx.cemi_handler.send_telegram.call_args_list == [
        call(write_address),
        call(read_address),
    ]
    await time_travel(3)
    with pytest.raises(ManagementConnectionError):
        await task
```
we can wait for the timeout after the requests were sent.